pwnlib/util/cyclic.py

```diff
@@ -344,3 +329,128 @@ def _gen_find(subseq, generator):
         if saved == subseq:
             return pos
     return -1
+
+def _join_sequence(seq, alphabet):
+    if isinstance(alphabet, six.text_type):
+        return ''.join(seq)
+    elif isinstance(alphabet, bytes):
+        return bytes(bytearray(seq))
+    else:
+        return seq
+
+class cyclic_gen(object):
+    """
+    Creates a stateful cyclic generator which can generate sequential chunks of de Bruijn sequences.
+
+    >>> g = cyclic_gen() # Create a generator
+    >>> g.get(4) # Get a chunk of length 4
+    b'aaaa'
+    >>> g.get(4) # Get a chunk of length 4
+    b'baaa'
+    >>> g.get(8) # Get a chunk of length 8
+    b'caaadaaa'
+    >>> g.get(4) # Get a chunk of length 4
+    b'eaaa'
+    >>> g.find(b'caaa') # Position 8, which is in chunk 2 at index 0
+    (8, 2, 0)
+    >>> g.find(b'aaaa') # Position 0, which is in chunk 0 at index 0
+    (0, 0, 0)
+    >>> g.find(b'baaa') # Position 4, which is in chunk 1 at index 0
+    (4, 1, 0)
+    >>> g.find(b'aaad') # Position 9, which is in chunk 2 at index 1
+    (9, 2, 1)
+    >>> g.find(b'aada') # Position 10, which is in chunk 2 at index 2
+    (10, 2, 2)
+    >>> g.get() # Get the rest of the sequence
+    b'faaagaaahaaaiaaajaaa...yyxzyzxzzyxzzzyyyyzyyzzyzyzzzz
```

Comment:
You missed the final quote
```suggestion
    b'faaagaaahaaaiaaajaaa...yyxzyzxzzyxzzzyyyyzyyzzyzyzzzz'
```

Reply:
Wow, how sloppy of me. Fixed.